pkg/alertmanager/validation/validation.go

```diff
@@ -33,3 +34,25 @@ func ValidateURL(url string) (*config.URL, error) {
 	}
 	return &u, nil
 }
+
+// ValidateSecretURL against config.URL
+// This is for URLs which are retrieved from secrets and should not
+// logged part of the err.
+func ValidateSecretURL(url string) error {
+	var u config.URL
+
+	err := json.Unmarshal([]byte(fmt.Sprintf(`"%s"`, url)), &u)
```

Comment:
Just out of curiosity, can't we just use `url.Parse()`? I see that it hasn't been used in the `ValidateURL()` func as well, is there any particular reason why we opt for manually un-marshalling the URL?

Reply:
My initial understanding was since `config.url` is a custom type exposed by the `alermanager` package which has custom methods implemented so we might be using this in-order to be able to call these methods as per need. 

But now I am seeing it also has `SecretURL` as type, not sure if we could used that in the  `validateSecretURL()` method.

https://github.com/prometheus/alertmanager/blob/main/config/config.go#L128-L150